src/redturtle/prenotazioni/behaviors/booking_folder/notification_email/events.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+from logging import getLogger
+
+from zope.component import getMultiAdapter
+from zope.globalrequest import getRequest
+
+from redturtle.prenotazioni import is_migration
+from redturtle.prenotazioni.interfaces import IBookingNotificationSender
+from redturtle.prenotazioni.interfaces import IPrenotazioneEmailMessage
+
+from .notification_email import INotificationEmail
+
+logger = getLogger(__name__)
+
+
+def booking_folder_provides_current_behavior(booking):
+    return INotificationEmail.providedBy(booking.getPrenotazioniFolder())
+
+
+def send_email_notification_on_transition(context, event) -> None:
+    if not booking_folder_provides_current_behavior(context):
+        return
+
+    booking_folder = context.getPrenotazioniFolder()
+    flags = {
```

Comment:
this flags logic is the same in each notification gateway..can we reuse it someway?
If we need to update the logic it's better to do it once.

Reply:
the code of the adapters is more or less the same for each behavior because the "core" is in sender and message adapters.

Can we have a base class with all of this logic, and set in specific ones only the interfaces needed to call the adapters?

Reply:
Moved the flags generation, but the rest even if it is very same code, is handling the different behaviours at this point. Also keeping in mind that plone is not following the DRY principe itself i would keep the rest so we can extend the code at any moment.

Reply:
I'm not convinced, but I'll mark it as resolved, and let's move on.